lib/shared-method.js

```diff
@@ -608,6 +614,27 @@ SharedMethod.prototype.isDelegateFor = function(suspect, isStatic) {
 };
 
 /**
+ * Determine if this shared method invokes the given "suspect" function.
+ *
+ * @example
+ * ```js
+ * sharedMethod.isDelegateForName(myClass.myMethod); // pass a function
+ * sharedMethod.isDelegateForName(myClass.prototype.myInstMethod);
```

Comment:
I think `suspect` arg should be always a string, shouldn't it?

Could you please add an `assert` as the first thing in the method to verify this precondition?